velox/common/memory/MallocAllocator.h

```diff
@@ -37,6 +37,17 @@ class MallocAllocator : public MemoryAllocator {
     }
   }
 
+  void registerCache(const std::shared_ptr<Cache>& cache) override {
+    VELOX_CHECK_NULL(cache_);
+    VELOX_CHECK_NOT_NULL(cache);
+    VELOX_CHECK(cache->allocator() == this);
+    cache_ = cache;
+  }
+
+  Cache* cache() const override {
+    return cache_.get();
```

Comment:
We can move cache_ to the base class and check zero allocated cache space in followup? Thanks!